### PR TITLE
Apimeta Set/RemoveStatusCondition: Indicate change

### PR DESCRIPTION
**Pull request by alvaroaleman**
The SetStatusCondition and RemoveStatusCondition currently do not indicate if they changed anything. In most cases that information is necessary to determine if an Update of the object is needed. This change adds a boolean return to them that indicate if they changed anything.

As the two functions had no return at all prior to this, this shouldn't break anything.

/sig api-machinery
/kind feature

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by alexzielenski — APPROVED** [review]
/lgtm

Thanks for adding it to tests. I don't see a problem with adding the return. I think usually a DeepEqual is used to decide if an update is necessary, but can see the merits of just being told directly if there was a change.

/assign @jpbetz

File: staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go [hunk 1]
```diff
@@ -27,17 +27,17 @@ import (
 //  1. if the condition of the specified type already exists (all fields of the existing condition are updated to
 //     newCondition, LastTransitionTime is set to now if the new status differs from the old status)
 //  2. if a condition of the specified type does not exist (LastTransitionTime is set to now() if unset, and newCondition is appended)
-func SetStatusCondition(conditions *[]metav1.Condition, newCondition metav1.Condition) {
+func SetStatusCondition(conditions *[]metav1.Condition, newCondition metav1.Condition) (changed bool) {
```
Comment (jpbetz) [inline staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go:30]: Update godoc to something like:

`// SetStatusCondition sets the corresponding condition in conditions to newCondition and returns true if the conditions are changed by this call.` ?

File: staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go [hunk 2]
```diff
@@ -47,18 +47,30 @@ func SetStatusCondition(conditions *[]metav1.Condition, newCondition metav1.Cond
...
+		changed = true
+	}
+
+	return changed
 }
 
 // RemoveStatusCondition removes the corresponding conditionType from conditions.
 // conditions must be non-nil.
-func RemoveStatusCondition(conditions *[]metav1.Condition, conditionType string) {
+func RemoveStatusCondition(conditions *[]metav1.Condition, conditionType string) (removed bool) {
```
Comment (jpbetz) [inline staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go:71]: Include information about return type in godoc here too?

**Review by jpbetz — CHANGES_REQUESTED** [review]
Request for expanded godoc then LGTM.

File: staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go (hunk 1 above)
Comment (alvaroaleman) [reply staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go:31]: Done

File: staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go (hunk 2 above)
Comment (alvaroaleman) [reply staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go:73]: Done, PTAL

**Pull request merged**